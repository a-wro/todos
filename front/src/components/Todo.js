import React from 'react'

export class Todo extends React.Component {
    state = {
        completed: this.props.completed,
    }

    handleClick = e => {
        this.setState({ completed: !this.state.completed })
    }

    render() {
        const { task } = this.props
        return (
            <div>
                <input
                    onClick={this.handleClick}
                    className="task-checkbox"
                    type="checkbox"
                    value={this.state.completed}
                />
                { task }
            </div>
        )
    }
}

export default Todo