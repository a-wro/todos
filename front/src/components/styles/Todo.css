.todo-btn {
    float: right;
    margin-right: 1.25rem;
    overflow: hidden;
}

.clearfix::after {
    content: "";
    clear: both;
    display: table;
}