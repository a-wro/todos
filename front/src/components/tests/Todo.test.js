import React from 'react'
import { Todo } from '../Todo'
import { shallow } from 'enzyme'


describe('Todo', () => {
    const props = { task: 'Test', completed: false}
    const todo = shallow(<Todo {...props} />)

    describe('Rendering', () => {
        it('renders properly', () => {
            expect(todo).toMatchSnapshot()
        })

        it('renders a checkbox', () => {
            expect(todo.find('.task-checkbox').exists()).toBe(true)
        })

    })

    describe('State related tests', () => {
        beforeEach(() => {
            todo.find('.task-checkbox').simulate('click')
        })

        it('Changes `completed` to true from previously false on click', () => {
            expect(todo.state().completed).toBe(true)
        })
    })

})
