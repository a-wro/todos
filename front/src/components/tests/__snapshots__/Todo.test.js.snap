// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Rendering renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Todo
    completed={false}
    task="Test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="task-checkbox"
          onClick={[Function]}
          type="checkbox"
          value={false}
/>,
        "Test",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "task-checkbox",
          "onClick": [Function],
          "type": "checkbox",
          "value": false,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "Test",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="task-checkbox"
            onClick={[Function]}
            type="checkbox"
            value={false}
/>,
          "Test",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "task-checkbox",
            "onClick": [Function],
            "type": "checkbox",
            "value": false,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "Test",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
